xrspatial/aspect.py

```diff
@@ -154,21 +154,52 @@ def _run_dask_numpy(data:da.Array) -> da.Array:
 
 
 def aspect(agg: xr.DataArray, name:str ='aspect'):
-    """Returns downward slope direction in compass degrees (0 - 360) with 0 at 12 o'clock.
+    """Calculates, for all cells in the array, the downward slope direction of each cell 
+        based on the elevation of its neighbors in a 3x3 grid. The value is measured 
+        clockwise in degrees with 0 and 360 at due north. Flat areas are given a value of -1.
 
     Parameters
     ----------
     agg : DataArray
+        Numpy, CuPy, Dask, or CuPy GPU data array with agg.data = elevation
 
     Returns
-    -------
+    ----------
     data: DataArray
+        Numpy, CuPy, Dask, or CuPy GPU data array with agg.out = aspect
```

Comment:
You may want to add some more details here. The output is a DataArray where its data dimension is the calculated aspect. Output data has the same type as of input. If we pass a DataArray with numpy-backed xarray, the output is also numpy-backed. Similarly with other input types.  Other attributes of the output aggregate are preserved as of input. Note that we don't have `.out` attribute in a DataArray.